### ISSUE TITLE
Default BeautifulSoup parser mangles DOCTYPEs

### ISSUE THREAD

**Issue by m-alger** [chromium/caterpillar#17]
Input this HTML file into `edit_code`:

```
﻿<!DOCTYPE html>
<html>
  <head>
  </head>
  <body>
  </body>
</html>
```

By default, BeautifulSoup will convert this into

```
<html>
 <body>
  <p>

   <!DOCTYPE html>
  </p>
 </body>
</html>
```

which is not valid HTML. Additionally, this then trips error #16.

**Comment by m-alger** [comment]
Resolved in [this CL](https://codereview.chromium.org/1614223002/).

**Comment by m-alger** [comment]
Resolved in 65eccfa030c7e89c511accf202fb4ccd4d2cb006.
